File: benchmarks/reference/java/StringStress.java
/** StringStress.java — String manipulation benchmark (50K iterations).
    Equivalent to examples/il/benchmarks/string_stress.il */
public class StringStress {
    public static void main(String[] args) {
        long sum = 0;
        for (long i = 0; i < 50000; i++) {
            String t = "Hello" + " " + "World" + "!";
            sum += t.length();
        }
        System.exit((int)(sum & 0xFF));
    }
}

File: benchmarks/reference/lua/string_stress.lua
-- string_stress.lua -- String manipulation benchmark (500K iterations).
-- Equivalent to examples/il/benchmarks/string_stress.il

local sum = 0
for i = 0, 499999 do
    local result = "Hello" .. " " .. "World" .. "!"
    sum = sum + #result
end

os.exit(sum & 0xFF)
